R/internals_RLum.R

```diff
@@ -1238,6 +1238,47 @@ SW <- function(expr) {
   return(TRUE)
 }
 
+#' @title Validate that a variable is not empty
+#'
+#' @param arg [character] (**required**): variable to validate.
+#' @param what [character] (**required**): the type of the variable, used
+#'        only in the message reported.
+#' @param throw.error [logical] (*with default*): whether an error should be
+#'        thrown in case of failed validation (`TRUE` by default). If `FALSE`,
+#'        the function raises a warning and proceeds.
+#' @param name [character] (*with default*): variable name to report in case
+#'        of error: if specified, it's reported as is; if not specified it's
+#'        inferred from the name of the variable tested and reported with
+#'        quotes.
+#'
+#' @return
+#' If `throw.error = TRUE`, the function throws an error and doesn't return
+#' anything. Otherwise, it will return a boolean to indicate whether validation
+#' was successful or not.
+#'
+#' @md
+#' @noRd
+.validate_not_empty <- function(arg, what, throw.error = TRUE,
+                                name = NULL) {
+
+  if (missing(what)) {
+    .throw_error("'what' must be provided")
+  }
```

Comment:
I think we don't need this because it is an internal function that is allowed to fail with standard R error messages.

Reply:
I've addressed your point as part of efb5f530473f87539862e670065756fb7a3a8175.